lib/filewatch/watch.rb

```diff
@@ -108,23 +130,34 @@ def each(&block)
             next
           end
 
+          if expired?(stat, watched_value)
+            if !watched_value.timeout_sent?
+              @logger.debug? && @logger.debug("#{path}: file expired")
+              yield(:timeout, path)
+              watched_value.timeout_sent = true
+            end
+            next
+          end
+
           inode = inode(path,stat)
-          if inode != @files[path][:inode]
-            @logger.debug? && @logger.debug("#{path}: old inode was #{@files[path][:inode].inspect}, new is #{inode.inspect}")
+          old_size = watched_value.size
+
+          if inode != watched_value.inode
+            @logger.debug? && @logger.debug("#{path}: old inode was #{watched_value.inode.inspect}, new is #{inode.inspect}")
             yield(:delete, path)
             yield(:create, path)
-          elsif stat.size < @files[path][:size]
-            @logger.debug? && @logger.debug("#{path}: file rolled, new size is #{stat.size}, old size #{@files[path][:size]}")
+          elsif stat.size < old_size
+            @logger.debug? && @logger.debug("#{path}: file rolled, new size is #{stat.size}, old size #{old_size}")
             yield(:delete, path)
             yield(:create, path)
-          elsif stat.size > @files[path][:size]
-            @logger.debug? && @logger.debug("#{path}: file grew, old size #{@files[path][:size]}, new size #{stat.size}")
+          elsif stat.size > old_size
+            @logger.debug? && @logger.debug("#{path}: file grew, old size #{old_size}, new size #{stat.size}")
             yield(:modify, path)
           end
 
-          @files[path][:size] = stat.size
-          @files[path][:inode] = inode
-        end # @files.keys.each
+          watched_value.size = stat.size
```

Comment:
Idea for this WatchedFile - what about using watched_value.update(stat) ? Maybe not... I'm mostly feeling a dislike of accessors lately, what do you think? Open to also deferring any changes based on this comment to a future PR